DateTimeUtils.dfy

```diff
@@ -0,0 +1,136 @@
+include "DateTimeConstant.dfy"
+
+module DateTimeUtils {
+  import opened Std.Strings
+  import opened DateTimeConstant
+
+  // Month names for better error messages
+  const MONTH_NAMES: seq<string> := ["January", "February", "March", "April", "May", "June",
+                                     "July", "August", "September", "October", "November", "December"]
+
+  // Function versions for use in function contexts
+  function {:extern "DateTimeImpl", "ToEpochTimeMilliseconds"}
```

Comment:
Good call using externs for these conversions!

I will leave another review tomorrow after I think on it some more, but so far I'm inclined to defer ANY calendar-sensitive operations to externs, to avoid even having to declare any kind of calendar concept in the Dafny for now.